File: dotnet/azure/Mcma.Azure.BlobStorage/BlobStorageLocator.cs
using Mcma.Core;

namespace Mcma.Azure.BlobStorage
{
    public abstract class BlobStorageLocator : IUrlLocator
    {
        /// <summary>
        /// Gets or sets the name of the storage account
        /// </summary>
        /// <value></value>
        public string StorageAccountName { get; }

        /// <summary>
        /// Gets or sets the share on which the file resides
        /// </summary>
        public string Container { get; set; }

        /// <summary>
        /// Gets the url to the file on Blob storage
        /// </summary>
        public abstract string Url { get; }
    }
}

File: azure/blob-storage/lib/blob-storage-locator.ts
import { Locator, LocatorProperties } from "@mcma/core";

export interface BlobStorageLocatorProperties extends LocatorProperties {
    storageAccountName: string;
    container: string;
}

export abstract class BlobStorageLocator extends Locator implements BlobStorageLocatorProperties {
    storageAccountName: string;
    container: string;

    protected constructor(type: string, properties: BlobStorageLocatorProperties) {
        super(type, properties);
        this.checkProperty("storageAccountName", "string");
        this.checkProperty("container", "string");
    }

    abstract get url(): string;
}
